### PR TITLE
feature/golang

### PR DESCRIPTION
**Pull request by larsks**
- Coalesce thoth pre-commit images
- Add go tooling to thoth pre-commit image

Addresses #38

### CODE REVIEW

File: thoth-precommit/Pipfile [hunk 1]
```diff
@@ -9,4 +9,4 @@ verify_ssl = true
 pre-commit = "*"
 
 [requires]
-python_version = "3.6"
+python_version = "3.9"
```
Comment (goern) [inline thoth-precommit/Pipfile:12]: if we change this, do we need to `pipenv relock` ?

Comment (larsks) [reply thoth-precommit/Pipfile:12]: You'll note that `Pipfile.lock` and `requirements.txt` are both updated (from running `pipenv lock` and `pipenv lock -r > requirements.txt`).

**Comment by larsks** [comment]
@goern does this need any more work?

**Comment by larsks** [comment]
There *is* more work to do: it turns out that the Fedora 34's glibc is too new, so binaries produced on this platform won't run on rhel/centos 7/8, fedora < 33, or Ubuntu < 20.10. The easiest solution is to base on the ubi8 image, which gets us go 1.15 and produces binaries that work on most current distributions.

The downside to go 1.15 is that it can't produce darwin-arm64 binaries, which are required by Apple's M1 hardware.

Do folks have any preference here?

**Comment by goern** [comment]
I'm +1 for ubi8.4+go1.15 it will also enable us to use py3.9 on ubi8.4

**Comment by larsks** [comment]
I'm trying to build a ubi38-py39 image, but I ran into this weirdness:

```
curl "https://raw.githubusercontent.com/thoth-station/s2i-thoth/master/ubi8-py39/requirements.txt" -o requirements.txt
```

Why are we copying in `requirements.txt` with curl rather than a `COPY` directive in the Dockerfile?  This makes it impossible to build the image before the code has landed because the URL won't exist. Will people yell at people if I just replace that with e.g:

```
COPY requirements.txt requirements.txt
```

**Comment by fridex** [comment]
@larsks I think this is a leftover from times when requirements files were not present in the build context. Feel free to raise a PR, if the solution work, it is much better than the `curl` hack. 👍🏻

**Review by harshad16 — CHANGES_REQUESTED** [review]
can we please not delete the `thoth-precommit-ubi-py38` and `thoth-precommit-ubi8` block.

**Comment by goern** [comment]
@larsks what is the status of this?

**Pull request closed without merging**